airbyte-integrations/connectors/source-iterable/source_iterable/iterable_streams.py

```diff
@@ -0,0 +1,204 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+import json
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple, Union
+
+import pendulum
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.http import HttpStream
+from pendulum.datetime import DateTime
+
+
+@dataclass
+class StreamSlice:
+    start_date: DateTime
+    end_date: DateTime
+
+
+class IterableStream(HttpStream, ABC):
+
+    # Hardcode the value because it is not returned from the API
+    BACKOFF_TIME_CONSTANT = 10.0
+    # define date-time fields with potential wrong format
+
+    url_base = "https://api.iterable.com/api/"
+    primary_key = "id"
+
+    def __init__(self, api_key, **kwargs):
+        super().__init__(**kwargs)
+        self._api_key = api_key
+
+    @property
+    @abstractmethod
+    def data_field(self) -> str:
+        """
+        :return: Default field name to get data from response
+        """
+
+    def backoff_time(self, response: requests.Response) -> Optional[float]:
+        return self.BACKOFF_TIME_CONSTANT
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        """
+        Iterable API does not support pagination
+        """
+        return None
+
+    def request_params(self, **kwargs) -> MutableMapping[str, Any]:
+        return {"api_key": self._api_key}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        records = response_json.get(self.data_field, [])
+
+        for record in records:
+            yield record
+
+
+class IterableExportStream(IterableStream, ABC):
+
+    cursor_field = "createdAt"
+    primary_key = None
+
+    def __init__(self, start_date, **kwargs):
+        super().__init__(**kwargs)
+        self._start_date = pendulum.parse(start_date)
+        self.stream_params = {"dataTypeName": self.data_field}
+
+    def path(self, **kwargs) -> str:
+        return "export/data.json"
+
+    def backoff_time(self, response: requests.Response) -> Optional[float]:
+        # Use default exponential backoff
+        return None
+
+    # For python backoff package expo backoff delays calculated according to formula:
+    # delay = factor * base ** n where base is 2
+    # With default factor equal to 5 and 5 retries delays would be 5, 10, 20, 40 and 80 seconds.
+    # For exports stream there is a limit of 4 requests per minute.
+    # Tune up factor and retries to send a lot of excessive requests before timeout exceed.
+    @property
+    def retry_factor(self) -> int:
+        return 20
```

Comment:
```suggestion
    retry_factor = 20
```
It is `property`, why can't we override how it's done for url_base or primary_key?

Reply:
Im not fan of this approach cause it replace property with a variable. If it works in python that doesn't mean it correct.
Alternative version could be:
```python
@IterableStream.retry_factor.getter
def retry_factor(self):
    return 20
```
But it seems to verbose for me

Reply:
I understand you, but it looks pretty funny that we override some of the `property` methods as a variable, and some as `property` methods.